### ISSUE TITLE
2.02 / 2.03 (Windows) Ethash (4GB) & KawPow (2GB): No hashrate for Ethash (with AMD RX5700 8GB)

### ISSUE THREAD

**Issue by UselessGuru** [sp-hash/TeamBlackMiner#389]
Edit: Same with 2.03 (see below)

[TeamBlackMiner-v2.02\TBMiner.exe --algo eth+rvn --hostname europe.ethash-hub.miningpoolhub.com --wallet ****************.**************3--port 20565 --server-passwd x --rvn-hostname kawpow.eu.mine.zpool.ca --rvn-wallet ************************ --rvn-passwd ***************,c=BTC,zap=GPN --rvn-port 1325 --api --api-version 1.4 --api-port 4001 --cl-devices [0]](url)

Note: Ethash epoch 276 is only 3.4GB. Rvn DAG is approx 1.05GB

```
00:00:00 [2023-10-08 20:35:50.027]
00:00:00 [2023-10-08 20:35:50.102] OpenCL driver version: 351.513
00:00:00 [2023-10-08 20:35:50.126] Cuda driver version: 12.2
00:00:00 [2023-10-08 20:35:50.126] Cuda runtime API version: 12.2
00:00:00 [2023-10-08 20:35:50.126]
00:00:00 [2023-10-08 20:35:50.126]
00:00:00 [2023-10-08 20:35:50.126]
00:00:00 [2023-10-08 20:35:50.126] Welcome to Team Black Miner 2.02
00:00:00 [2023-10-08 20:35:50.126]
00:00:00 [2023-10-08 20:35:50.126]
00:00:00 [2023-10-08 20:35:50.126] GPU0 gfx1010:xnack- 8GB added to miner as OpenCL device
00:00:00 [2023-10-08 20:35:50.126] Trying kawpow.eu.mine.zpool.ca:1325 (kawpow)
00:00:00 [2023-10-08 20:35:50.126] Trying europe.ethash-hub.miningpoolhub.com:20565 (ethash)
00:00:00 [2023-10-08 20:35:50.126] Api server started at 127.0.0.1:4001
00:00:00 [2023-10-08 20:35:50.134] GPU0 Started CL work thread
00:00:00 [2023-10-08 20:35:50.292] europe.ethash-hub.miningpoolhub.com: Pool responded successfully to subscribe
00:00:00 [2023-10-08 20:35:50.326] kawpow.eu.mine.zpool.ca: Pool responded to subscribe
00:00:00 [2023-10-08 20:35:50.419] europe.ethash-hub.miningpoolhub.com: Difficulty is set to 3.000000
00:00:00 [2023-10-08 20:35:50.450] GPU0 Starting generating zil dag (Epoch: 1)
00:00:00 [2023-10-08 20:35:50.514] europe.ethash-hub.miningpoolhub.com: Pool responded successfully to authorize
00:00:00 [2023-10-08 20:35:50.672] kawpow.eu.mine.zpool.ca: Pool responded successfully to extranonce subscribe
00:00:00 [2023-10-08 20:35:50.688] kawpow.eu.mine.zpool.ca: Pool responded successfully to authorize
00:00:00 [2023-10-08 20:35:52.225] Job 2c36
00:00:00 [2023-10-08 20:35:53.217] GPU0 generating kawpow DAG (Epoch: 30)
00:00:00 [2023-10-08 20:35:53.309] GPU0 DAG generated in 2.86 seconds (1056767kb)
00:00:00 [2023-10-08 20:35:58.127] GPU0 kawpow dag generated (Epoch: 30)
00:00:00 [2023-10-08 20:35:58.127] GPU0 Starting generating ethash dag (Epoch: 276)
00:00:01 [2023-10-08 20:36:50.040] miningpoolhub.com (ethash)    PING: 69ms    DIFFICULTY: 3.00     EPOCH: 276
00:00:01 [2023-10-08 20:36:50.040]
00:00:01 [2023-10-08 20:36:50.040] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:01 [2023-10-08 20:36:50.040] GPU0    5700XT   OpenCL  0       0       0/0     46      2040    1840    0
00:00:01 [2023-10-08 20:36:50.040]                                                                          0
00:00:01 [2023-10-08 20:36:50.040] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:01 [2023-10-08 20:36:50.040] GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:01 [2023-10-08 20:36:50.040]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
```

**Comment by UselessGuru** [comment]
Still the same with 2.03.

Miner never completes buiilding DAG for Ethash:
```
00:00:00 [2023-10-10 07:44:57.604]
00:00:00 [2023-10-10 07:44:57.772] OpenCL driver version: 537.42
00:00:00 [2023-10-10 07:44:57.804] Cuda driver version: 12.2
00:00:00 [2023-10-10 07:44:57.804] Cuda runtime API version: 12.2
00:00:00 [2023-10-10 07:44:57.804]
00:00:00 [2023-10-10 07:44:57.804]
00:00:00 [2023-10-10 07:44:57.804]
00:00:00 [2023-10-10 07:44:57.804] Welcome to Team Black Miner 2.03
00:00:00 [2023-10-10 07:44:57.804]
00:00:00 [2023-10-10 07:44:57.804]
00:00:00 [2023-10-10 07:44:57.804] GPU0 gfx1010:xnack- 8GB added to miner as OpenCL device
00:00:00 [2023-10-10 07:44:57.823] GPU0 Started CL work thread
00:00:00 [2023-10-10 07:44:57.823] Api server started at 127.0.0.1:4001
00:00:00 [2023-10-10 07:44:57.823] Trying europe.ethash-hub.miningpoolhub.com:20565 (ethash)
00:00:00 [2023-10-10 07:44:57.825] Trying kawpow.eu.mine.zpool.ca:1325 (kawpow)
00:00:00 [2023-10-10 07:44:58.009] europe.ethash-hub.miningpoolhub.com: Pool responded successfully to subscribe
00:00:00 [2023-10-10 07:44:58.107] europe.ethash-hub.miningpoolhub.com: Difficulty is set to 3.000000
00:00:00 [2023-10-10 07:44:58.131] kawpow.eu.mine.zpool.ca: Pool responded to subscribe
00:00:00 [2023-10-10 07:44:58.140] GPU0 Starting generating zil dag (Epoch: 1)
00:00:00 [2023-10-10 07:44:58.215] europe.ethash-hub.miningpoolhub.com: Pool responded successfully to authorize
00:00:00 [2023-10-10 07:44:58.476] kawpow.eu.mine.zpool.ca: Pool responded successfully to extranonce subscribe
00:00:00 [2023-10-10 07:44:58.480] kawpow.eu.mine.zpool.ca: Pool responded successfully to authorize
00:00:00 [2023-10-10 07:45:00.735] Job 1487
00:00:00 [2023-10-10 07:45:02.233] GPU0 generating kawpow DAG (Epoch: 58)
00:00:00 [2023-10-10 07:45:03.597] GPU0 DAG generated in 5.46 seconds (1056767kb)
00:00:00 [2023-10-10 07:45:09.230] GPU0 kawpow dag generated (Epoch: 58)
00:00:00 [2023-10-10 07:45:09.492] GPU0 Starting generating ethash dag (Epoch: 276)
00:00:00 [2023-10-10 07:45:12.662] KAWPOW: GPU0 Accepted (1/1/0) Reply: 169ms
00:00:01 [2023-10-10 07:45:58.033] miningpoolhub.com (ethash)    PING: 81ms    DIFFICULTY: 3.00     EPOCH: 276
00:00:01 [2023-10-10 07:45:58.033]
00:00:01 [2023-10-10 07:45:58.033] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:01 [2023-10-10 07:45:58.033] GPU0    5700XT   OpenCL  0       0       0/0     46      2040    1840    0
00:00:01 [2023-10-10 07:45:58.033]                                                                          0
00:00:01 [2023-10-10 07:45:58.033] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:01 [2023-10-10 07:45:58.033] GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:01 [2023-10-10 07:45:58.033]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:01 [2023-10-10 07:45:58.033]
00:00:01 [2023-10-10 07:45:58.033] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
```

**Comment by UselessGuru** [comment]
Still the same with 2.04

**Comment by UselessGuru** [comment]
Fixed in 2.05
